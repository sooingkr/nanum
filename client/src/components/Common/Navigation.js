import React, { Component } from "react";
import { connect } from 'react-redux';
import { withRouter, Link } from 'react-router-dom';
import { PropTypes } from "prop-types";

import { 
  Navbar, 
  Nav, 
  NavItem, 
  NavDropdown, 
  Image, 
  MenuItem,
  Collapse,
} from "react-bootstrap";

import FoodSearchBoxContainer from '../../containers/FoodSearch/FoodSearchBoxContainer';

import {HomeDuck} from '../../containers/Home/HomeDuck';

class Navigation extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showNavMenu: false,
    }
  }

  toggleNav = () => {
    this.setState({ showNavMenu: !this.state.showNavMenu });
  };

  handleHaccpButton = () => {
    const { toggleChatBox, history } = this.props;
    const userId = '';

    if (userId) { /* user logged in */
      toggleChatBox();

    } else { /* not logged in */
      history.push('/service');

    }
  };

  render() {
    const { history } = this.props;

    return (
      <header>
        <Navbar default collapseOnSelect>
          <div className="navigation__logo">
            <Link to="/" title="신 안전먹거리" className='logo'>
              <Image className="logo-circle" responsive src="https://fresh.ihaccp.or.kr/images/fresh/common/header_icon01.gif" alt="Fresh"/>
              <Image className="logo-fresh" responsive src="https://fresh.ihaccp.or.kr/images/fresh/common/header_logo02.gif" alt="Fresh"/>
            </Link>
            <button className="navigation__toggle" 
                    type="button"
                    onClick={this.toggleNav}>
              <i className="fa fa-caret-down" aria-hidden="true"></i>
            </button>
          </div>

          <div className="navigation__main">
            <Collapse in={this.state.showNavMenu}>
              <Nav pullRight>
                <NavItem onClick={navigateHelper('/', history)} title="추천먹거리정보" eventKey={1}>추천먹거리정보</NavItem>
                <NavItem onClick={this.handleHaccpButton} title="HACCP 기술상담" eventKey={2}>HACCP 기술상담</NavItem>
                <NavItem onClick={navigateHelper('/safety/searchProduct.do', history)} title="안전먹거리정보" eventKey={3}>안전먹거리정보</NavItem>
                <NavItem onClick={navigateHelper('/board/boardList.do?board=21', history)} title="민원" eventKey={4}>민원</NavItem>
                <NavItem onClick={navigateHelper('/lod/info.do', history)} title="데이터활용" eventKey={5}>데이터활용</NavItem>
                <NavItem onClick={navigateHelper('/product/2', history)} title="데이터활용" eventKey={6}>데이터활용</NavItem>
              </Nav>
            </Collapse>
          </div> 

          <div className="navigation__secondary">
            <Nav>
              <FoodSearchBoxContainer />
              <NavDropdown eventKey={7} title="" id="secondary-nav-dropdown">
                <MenuItem eventKey={7.1} onClick={navigateHelper('/login', history)}>Login</MenuItem>
                <MenuItem eventKey={7.2} onClick={navigateHelper('/dashboard', history)}>Dashboard</MenuItem>
              </NavDropdown>
            </Nav>
          </div>
        </Navbar>
      </header>
    );
  }
} 

export default withRouter(
  connect(null, {
    toggleChatBox: HomeDuck.actions.toggleChatBox,
  })(Navigation)
);

Navigation.propTypes = {
  toggleNav: PropTypes.func,
  handleHaccpButton: PropTypes.func,
  navigateHelper: PropTypes.func,
  toggleChatBox: PropTypes.func,
  history: PropTypes.object.isRequired,
}

function navigateHelper(path, history) {
  return function() {
    history.push(path);
  }
}