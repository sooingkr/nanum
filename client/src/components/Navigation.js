import React, { Component } from "react";
import { withRouter, Link } from 'react-router-dom';
import { 
  Navbar, 
  Nav, 
  NavItem, 
  NavDropdown, 
  Image, 
  MenuItem,
  Collapse,
} from "react-bootstrap";
import FoodSearchBoxContainer from '../containers/FoodSearch/FoodSearchBoxContainer';

class Navigation extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showNavMenu: false,
    }
  }

  toggleNav = () => {
    this.setState({ showNavMenu: !this.state.showNavMenu });
  }

  render() {
    const { history } = this.props;

    return (
      <header>
        <Navbar default collapseOnSelect>
          <div className="navigation__logo">
            <Link to="/" title="신 안전먹거리" className='logo'>
              <Image 
                src="https://images-na.ssl-images-amazon.com/images/G/01/omaha/images/yoda/logos/fresh-modal-3x._CB315803244_.png" 
                alt="Fresh" 
                className="img-responsive"
              />
            </Link>
            <button className="navigation__toggle" 
                    type="button"
                    onClick={this.toggleNav}>
              <i className="fa fa-caret-down" aria-hidden="true"></i>
            </button>
          </div>

          <div className="navigation__main">
            <Collapse in={this.state.showNavMenu}>
              <Nav pullRight>
                <NavItem onClick={navigateHelper('/', history)} title="신 안전먹거리" eventKey={1}>신 안전먹거리</NavItem>
                <NavItem onClick={navigateHelper('/safety/searchProduct.do', history)} title="구 안전먹거리" eventKey={2}>구 안전먹거리</NavItem>
                <NavItem onClick={navigateHelper('/company/companyMain.do', history)} title="HACCP관리 전산기준서" eventKey={3}>HACCP관리 전산기준서</NavItem>
                <NavItem onClick={navigateHelper('/board/boardList.do?board=21', history)} title="민원" eventKey={4}>민원</NavItem>
                <NavItem onClick={navigateHelper('/lod/info.do', history)} title="데이터활용" eventKey={5}>데이터활용</NavItem>
                <NavItem onClick={navigateHelper('/product/2', history)} title="데이터활용" eventKey={6}>데이터활용</NavItem>
              </Nav>
            </Collapse>
          </div> 

          <div className="navigation__secondary">
            <Nav>
              <FoodSearchBoxContainer />
              <NavDropdown eventKey={7} title="" id="secondary-nav-dropdown">
                <MenuItem eventKey={7.1} onClick={navigateHelper('/login', history)}>Login</MenuItem>
                <MenuItem eventKey={7.2} onClick={navigateHelper('/dashboard', history)}>Dashboard</MenuItem>
              </NavDropdown>
            </Nav>
          </div>
        </Navbar>
      </header>
    );
  }
} 

export default withRouter(Navigation);

function navigateHelper(path, history) {
  return function() {
    history.push(path);
  }
}