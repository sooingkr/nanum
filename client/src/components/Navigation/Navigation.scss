@import "../../stylesheets/base/_variables.scss";
@import "../../stylesheets/base/common.scss";
@import "../../stylesheets/mixin/mixin.scss";

.App {
  .navbar {
    background-color: $white;
    background-image: none;
    border: none;
    height: 76px;
    margin-bottom: 0;
    .logo-block {
      margin-top: 17px;
    }
    .logo {
      float: left;
      width: 107px;
      height: 42px;
      margin-top: 5px;
    }
    .dropdown-toggle {
      background-color: transparent;
      background-image: none;
      border: none;
      box-shadow: none;
      -webkit-box-shadow: none;
      -moz-box-shadow: none;

      &:before {
        font-family: 'Glyphicons Halflings';
        content: "\e008";
      }
    }
    .caret {
      display: none;
    }
  }
}

@media screen and (min-width: $breakpoints-tablet) {
  .App {
    .navbar {
      .navbar-brand, .navbar-nav {
        position: absolute;
        top: 16px;
      }
      .navbar-nav {
        &:last-child {
          right: 0;
        }
      }
    }
  }
}

@media screen and (max-width: $breakpoints-phone) {
  .App {
    .navbar {
      .navbar-header {
        height: 76px;

        .navbar-toggle {
          float: none;
          background-color: transparent;
          border: none;
          &:before {
            font-family: 'Glyphicons Halflings';
            content: "\e252";
          }
          .icon-bar {
            display: none;
          }
        }
      }
      .search-box {
        top: 28px;
        left: 37%;
        width: 63%;

        .form-control, .btn-default {
          position: absolute;
          top: -18px;
        }
        .btn-default {
          right: 0;
        }
      }
      .navbar-collapse {
        background-color: $white;
        position: absolute;
        top: 0;
        left: 0;
        right: 0;
        z-index: 9999;
        min-height: 360px;
      }
    }
  }
}
