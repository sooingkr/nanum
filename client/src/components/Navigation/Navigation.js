import React from "react";
import { Row, Col, Navbar, Nav, NavItem, Image, ButtonToolbar, Dropdown, MenuItem, Glyphicon} from "react-bootstrap";

import SearchBox from "../../components/SearchBox/SearchBox.js";

import "./Navigation.scss";
import { logo } from '../../assets/images/logo.svg';

const Navigation = () => (
  <header>
    <Navbar default collapseOnSelect>
      <Row>
        <Col md={5}>
          <Navbar.Header>
            <Navbar.Brand>
              <a className="pull-left" href="/" title="Home"><Image src={logo} alt="Fresh" className="img-responsive"/></a>
            </Navbar.Brand>
            <Navbar.Toggle />
          </Navbar.Header>
        </Col>
        <Col md={7}>
          <Navbar.Collapse>
            <Nav>
              <NavItem href="/" title="" eventKey={1}>Link</NavItem>
              <NavItem href="/dashboard" title="Dashboard" eventKey={2}>Dashboard</NavItem>
            </Nav>
            <Nav pullRight>
              <NavItem><SearchBox typeName=" "/></NavItem>
              <NavItem>
                <ButtonToolbar>
                  <Dropdown id="dropdown-user">
                    <Dropdown.Toggle>
                      <Glyphicon glyph="user" />
                    </Dropdown.Toggle>
                    <Dropdown.Menu className="super-colors">
                      <MenuItem eventKey="1">Action</MenuItem>
                      <MenuItem eventKey="2">Another action</MenuItem>
                      <MenuItem eventKey="3" active>Active Item</MenuItem>
                      <MenuItem divider />
                      <MenuItem eventKey="4">Separated link</MenuItem>
                    </Dropdown.Menu>
                  </Dropdown>
                </ButtonToolbar>
              </NavItem>
            </Nav>
          </Navbar.Collapse>
        </Col>
      </Row>
    </Navbar>
  </header>
)

export default Navigation;