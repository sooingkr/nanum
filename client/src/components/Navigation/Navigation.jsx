import React from "react";
import { Navbar, Nav, NavItem, Image } from "react-bootstrap";

import "./Navigation.scss";
import { logo } from '../../assets/images/logo.svg';

export const Navigation = () => {
  return (
    <header>
      <Navbar default collapseOnSelect>
        <Navbar.Header>
          <Navbar.Brand>
            <a href="/" title="Home"><Image src={logo} alt="Fresh" className="img-responsive"/></a>
          </Navbar.Brand>
        </Navbar.Header>
        <Navbar.Collapse>
          <Nav>
            <NavItem eventKey={1} href="#">Link</NavItem>
            <NavItem href="/dashboard" title="Dashboard" eventKey={1}>Dashboard</NavItem>
            <NavItem href="/" title="Link" eventKey={2}>Link</NavItem>
          </Nav>
        </Navbar.Collapse>
      </Navbar>
    </header>
  );
}
