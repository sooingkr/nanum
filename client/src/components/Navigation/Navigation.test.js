import React from 'react';
import { shallow } from 'enzyme';
import Navigation from "./Navigation";

describe('<Navigation />', () => {
  it('should render <Navigation /> components', () => {
    const wrapper = shallow(<Navigation />);
    expect(wrapper).toMatchSnapshot();
  });
});