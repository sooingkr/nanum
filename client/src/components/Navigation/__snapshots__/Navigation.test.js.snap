// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should render <Navigation /> components 1`] = `
<Route
  render={[Function]}
/>
`;
